src/components/calcite-alert/calcite-alert.tsx

```diff
@@ -57,19 +59,24 @@ export class CalciteAlert {
     : null;
 
   /** Color for the alert (will apply to top border and icon) */
-  @Prop({ reflect: true }) color: "blue" | "green" | "red" | "yellow" = "blue";
+  @Prop({ reflect: true }) color: CalciteStatusColor = "blue";
 
-  /** specify if the alert should display an icon */
-  @Prop() icon = false;
+  /** when used as a boolean set to true, show a default recommended icon. You can
+   * also pass a calcite-ui-icon name to this prop to display a requested icon */
+  @Prop({ reflect: true }) icon: string | boolean;
+
+  @Watch("icon") iconWatcher(): void {
```

Comment:
Does this update the icon if I use `<calcite-alert icon color="blue">` and then change to `<calcite-alert icon color="green">`? I think you would need to watch both `icon` and `color` props. May be easier to just do this in the render like:

```
const icon = setRequestedIcon(StatusIcons, this.icon, this.color)
```

If you keep this as a prop, watchers have their own section in the file order, IIRC. and The requested icon may need to be a `@State` element to render after update, but I could be wrong about that?

Reply:
I think like this:

```ts
@State() requestedIcon: string;

@Watch("icon")
@Watch("color")
updateRequestedIcon(): string {
  this.requestedIcon = setRequestedIcon(StatusIcons, this.icon, this.color);
}

```

Reply:
I'll add a watch for color too. Not sure about file order - all the component files just list the watcher after the prop. Although if they are combined here I can just place at the end of the prop list... or create a // Watchers file section?